### PR TITLE
fix(permissions): fix notifications permissions on firefox

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: test/permissions.spec.js [hunk 1]
```diff
@@ -52,19 +52,20 @@ module.exports.describe = function({testRunner, expect, WEBKIT, FFOX}) {
       await context.grantPermissions(['geolocation'], { origin: server.EMPTY_PAGE });
       expect(await getPermission(page, 'geolocation')).toBe('granted');
     });
-    it.fail(FFOX)('should grant notifications permission when listed', async({page, server, context}) => {
+    it('should grant notifications permission when listed', async({page, server, context}) => {
       await page.goto(server.EMPTY_PAGE);
       await context.grantPermissions(['notifications'], { origin: server.EMPTY_PAGE });
+      await page.waitFor(3000);
```
Comment (dgozman) [inline test/permissions.spec.js]: whoopsy

**Review by dgozman — COMMENTED** [review]
I think browser patch has a bug, so I wonder whether the flakiness is still there? 😄

File: test/permissions.spec.js (hunk 1 above)
Comment (pavelfeldman) [reply test/permissions.spec.js]: Removed

**Review by pavelfeldman — COMMENTED** [review]
> I think browser patch has a bug, so I wonder whether the flakiness is still there?

My patch is not fixing flakiness, it makes notifications permissions work. If you follow the code, you can see that exception on the content side was the thing that 'fixed' the flakiness in the first place. The bug you saw in the browser side was me making last-minute call beautification.

Since then I reworked the browser-side patch to only do the empty rountrip, it turned out that the `perm-change` event that it awaited never arrived.

**Review by dgozman — APPROVED** [review]

**Pull request merged**